src/redux/profile/profileMiddleware.ts

```diff
@@ -0,0 +1,42 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+
+import { api } from '@/api/api';
+import { onOpenConnection } from '@/redux/connection/connectionSlice';
+import { AppStartListening } from '@/redux/listenerMiddleware';
+
+import { setMenuItemVisible } from '../local/localSlice';
+
+export interface ProfileState {
+  uiPanelVisibility: {
+    // The key here should match the id of the menu item,
+    // else will be ignored
+    [key: string]: boolean;
+  };
+  markNodes: string[];
+}
+
+export const getProfile = createAsyncThunk('profile/getProfile', async () => {
+  const topic = api.startTopic('profile', {});
+  const { value } = await topic.iterator().next();
+  topic.cancel();
+  return value as ProfileState;
+});
+
+export const addProfileListener = (startListening: AppStartListening) => {
+  startListening({
+    actionCreator: onOpenConnection,
+    effect: async (_, listenerApi) => {
+      listenerApi.dispatch(getProfile());
+    }
+  });
+
+  startListening({
+    actionCreator: getProfile.fulfilled,
+    effect: async (action, listenerApi) => {
+      Object.entries(action.payload.uiPanelVisibility).forEach(([key, value]) => {
+        listenerApi.dispatch(setMenuItemVisible({ id: key, visible: value }));
+      });
```

Comment:
Do we want to do something explicitly here if the ID from OpenSpace is invalid? It will be silently ignored in `setMenuItemVisible`, so if a user has changed the json file there is no way to tell unless they later try to load that file using the load/save buttons